lean4/deps/smtlib/src/SMTLIB/Syntax.lean

```diff
@@ -686,11 +714,23 @@ def name_term (name : String) : forall {s : sort} (tm : term s), smtM (term s)
 
 def assert (b : term smt_bool) : smtM Unit := 
   modify (fun st => {st with script := (Raw.command.assert b) :: st.script })
+def setLogic (l : Raw.logic) : smtM Unit :=
+  modify (fun st => {st with script := (Raw.command.setLogic l) :: st.script })
+def setOption (o : Raw.option) : smtM Unit :=
+  modify (fun st => {st with script := (Raw.command.setOption o) :: st.script })
+def checkSatAssuming (bs : List (term smt_bool)) : smtM Unit :=
+  modify (fun st => {st with script := (Raw.command.checkSatAssuming bs) :: st.script })
+def exit : smtM Unit :=
+  modify (fun st => {st with script := Raw.command.exit :: st.script })
 
 def ex1 : smtM Unit :=
   do f <- declare_fun "f" [smt_bool, smt_bool] smt_bool;
      assert (f true false)
 
+def liftCommand (c : command) : smtM Unit :=
```

Comment:
@simonjwinwood this felt odd to add but I threw it in at least for the short term =\ there's probably a cleaner way to combine the SMT types/context/etc and the VCG stuff.